cvmfs/catalog_mgr_impl.h

```diff
@@ -876,9 +878,13 @@ void AbstractCatalogManager<CatalogT>::StageNestedCatalog(
 
     LogCvmfs(kLogCatalog, kLogDebug, "staging nested catalog at %s (%s)",
              i->mountpoint.c_str(), i->hash.ToString().c_str());
-    StageNestedCatalogByHash(i->hash, i->mountpoint);
-    break;
+    shash::Any hash       = shash::Any(i->hash);
+    PathString mountpoint = PathString(i->mountpoint);
+    Unlock();
```

Comment:
If I see it correctly `Unlock()` is called here in a for loop? I dont think `Unlock()` should be called multiple times while in the former implementation it was only called once.

The underlying `pthread_rwlock_unlock` has the following description:
The pthread_rwlock_unlock() function shall release a lock held on the read-write lock object referenced by rwlock. Results are **undefined** if the read-write lock rwlock is not held by the calling thread.

Reply:
Looks correct to me. At this point the code path exists the for loop/method.